CHANGELOG.md

```diff
@@ -0,0 +1,246 @@
+the commit tree of the [v3.0.1](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v3.0.1) release. [v3.1.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v3.1.0) is **NOT** a released
+version.  
+
+## [3.0.1](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/v3.0.1) (May 10, 2018)
+
+Bugfixes:
+
+  - [#167](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/167), [#168](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/168) Fixed issues reported by Coverity Scan.
+  - [#177](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/177) Fixes a memory corruption bug and handling of timeouts.
+
+Other:
+
+  - Add .travis.yml for Travis CI.
+  - Removed C++ sample.
+  - Removed includes of `inttypes.h`, which doesn't exist on some systems.
+  - [#175](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/175) Added comments on static allocation of MQTT topics.
+
+## [3.0.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/v3.0.0) (Apr 17, 2018)
+
+Bugfixes:
+
+  - [#152](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/152) Fixes potential buffer overflows in `parseStringValue` by requiring a size parameter in `jsonStruct_t`.
+  - [#155](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/155) Fixes other memory corruption bugs; also improves stability.
+
+The two bug fixes above are not backwards compatible with v2.3.0.  Please see [README.md](README.md#migrating-from-2x-to-3x) for details on migrating to v3.0.0.
+
+## [2.3.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/v2.3.0) (Mar 21, 2018)
+
+New Features:
+
+  - Add [AWS IoT Jobs](https://docs.aws.amazon.com/iot/latest/developerguide/iot-jobs.html) support.
+  - Use AWS IoT Core support for MQTT over port 443. MQTT connection now defaults to port 443.
+
+Pull requests:
+
+  - [#124](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/124) - Thing Shadow: Fix potential shadow buffer overflow
+  - [#135](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/135) - mbedtls_wrap: Fix unintialized variable usage
+  - Fix bugs in long-running integration tests.
+
+## [2.2.1](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/v2.2.1) (Dec 26, 2017)
+
+Bugfixes:
+
+  - [#115](https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/115) - Issue with new client metrics
+
+Pull requests:
+
+  - [#112](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/112) - Initialize msqParams.isRetained to 0 in publishToShadowAction()
+  - [#118](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/118) - mqttInitParams.mqttPacketTimeout_ms initialized
+
+## [2.2.0](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/v2.2.0) (Nov 22, 2017)
+
+New Features:
+
+  - Added SDK metrics string into connect packet
+
+Bugfixes:
+
+  - [#49](https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/49) - Add support for SHADOW_JSON_STRING as supported value
```

Comment:
do we need this?

It seems to me that if we list all this, then we need to move to github to all our issues tracking.

Reply:
This is the content of the [Change Log file from the v3.x series](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v3.0.1/CHANGELOG.md) as-is. I think there is value in keeping the content of the previous releases untouched.

Reply:
Ok, if this is from last release, I agree to leave it unchanged.